### PR TITLE
docs: Improve CLI usage wording

### PR DESCRIPTION
**Pull request by lornajane**
## What/Why/How?

This is a little light updating for both the CLI help/usage output and the documentation for the CLI. The aim is to add clarity, remove old options, and I added a page on the completion script.

## Check yourself

- [ ] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [ ] All new/updated code is covered with tests

## Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### CODE REVIEW

**Comment by lornajane** [comment]
Small pull request, but it needs both a fact check from @Redocly/dark-side and a proofread from @Redocly/technical-writers. Thanks, all!

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ yargs
   )
 
   .command(
-    'push [api] [maybeDestination] [maybeBranchName]',
```
Comment (IgorKarpiuk) [inline packages/cli/src/index.ts:131]: Yargs will not parse positional arguments without it. We need it for backward compatibility.

**Comment by tatomyr** [comment]
Please update e2e snapshots with `npm run e2e -- -u` and run prettier.

File: packages/cli/src/index.ts (hunk 1 above)
Comment (lornajane) [reply packages/cli/src/index.ts]: Ah, got it - and fixed. Thanks!

**Comment by lornajane** [comment]
Rebased after a bunch of other changes on `main`

**Review by IgorKarpiuk — APPROVED** [review]

File: docs/commands/completion.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# `completion`
+
+## Introduction
+
+The `completion` command generates the scripts to add autocompletion to you bash or zsh shell. This is a handy productivity boost if you regularly use `redocly` from the command line.
```
Comment (HCloward) [inline docs/commands/completion.md:5]:
```suggestion
The `completion` command generates the scripts to add autocompletion to your bash or zsh shell. This addition is a handy productivity boost if you regularly use `redocly` from the command line.
```

File: docs/commands/lint.md [hunk 3]
```diff
@@ -22,19 +24,19 @@ redocly lint --version
...
+| Option                 | Type     | Description                                                                                                                                           |
+| ---------------------- | -------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| apis                   | [string] | Array of API definition filenames that need to be linted. See [the Apis section](#apis) for more options.                                             |
+| --config               | string   | Specify path to the [configuration file](#custom-configuration-file).                                                                                 |
+| --extends              | [string] | [Extend a specific configuration](#extend-configuration) (defaults or config file settings).                                                          |
+| --format               | string   | Format for the output.<br />**Possible values:** `codeframe`, `stylish`, `json`, `checkstyle`, `codeclimate`, `summary`. Default value is `codeframe` |
```
Comment (HCloward) [inline docs/commands/lint.md:32]:
```suggestion
| --format               | string   | Format for the output.<br />**Possible values:** `codeframe`, `stylish`, `json`, `checkstyle`, `codeclimate`, `summary`. Default value is `codeframe`. |
```

File: docs/commands/login.md [hunk 4]
```diff
@@ -2,23 +2,22 @@
...
 Also, you can access your members-only (private) API definitions in the Redocly registry, and use the [`push`](./push.md) command.
 
 If you're having issues with the `login` command, use the `--verbose` option to display a detailed error trace (if any).
 
-
 ## Usage
 
+:::warning Note
+
+Go ahead and [generate a personal API key](../../settings/personal-api-keys.md), this is needed to log in.
```
Comment (HCloward) [inline docs/commands/login.md:17]:
```suggestion
Go ahead and [generate a personal API key](../../settings/personal-api-keys.md); this key is needed to log in.
```

File: docs/commands/push.md [hunk 5]
```diff
@@ -76,21 +76,21 @@ redocly push [-u] [--job-id id] [--batch-size number] <path/to/definition.yaml>
...
+| --batch-size     |  number  | Number of CI pushes expected within one batch. See [the Batch Size section](#batch-size) for more information.                                                                                                                                        |
+| --help           | boolean  | Help output for the command.                                                                                                                                                                                                                          |
+| --public         | boolean  | Make API definitions publicly accessible from the API Registry. Read more about [using the public option](#public).                                                                                                                                   |
+| --region,-r      |  string  | Which region to use when logging in. Supported values: `us`, `eu`. The `eu` region is limited to enterprise customers. Default value is `us`. Alternatively, set an environment variable `REDOCLY_DOMAIN` with the value the appropriate Redocly API. |
```
Comment (HCloward) [inline docs/commands/push.md:89]:
```suggestion
| --region,-r      |  string  | Which region to use when logging in. Supported values: `us`, `eu`. The `eu` region is limited to enterprise customers. Default value is `us`. Alternatively, set an environment variable `REDOCLY_DOMAIN` with the value of the appropriate Redocly API. |
```

File: packages/cli/src/index.ts [hunk 6]
```diff
@@ -431,40 +430,41 @@ yargs
...
           t: {
             alias: 'template',
-            describe: 'Path to handlebars page template, see https://git.io/vh8fP for the example',
+            describe: 'Path to handlebars page template, see https://git.io/vh8fP for the example.',
             type: 'string',
           },
           templateOptions: {
             describe:
-              'Additional options that you want pass to template. Use dot notation, e.g. templateOptions.metaDescription',
+              'Additional options to pass to the template. Use dot notation, e.g. templateOptions.metaDescription',
```
Comment (HCloward) [inline packages/cli/src/index.ts:460]:
```suggestion
              'Additional options to pass to the template. Use dot notation, for example, templateOptions.metaDescription',
```

File: packages/cli/src/index.ts [hunk 7]
```diff
@@ -431,40 +430,41 @@ yargs
...
           },
           theme: {
-            describe: 'Redoc theme.openapi, use dot notation, e.g. theme.openapi.nativeScrollbars',
+            describe:
+              'Redoc theme.openapi configuration. Use dot notation, e.g. theme.openapi.nativeScrollbars',
```
Comment (HCloward) [inline packages/cli/src/index.ts:464]:
```suggestion
              'Redoc theme.openapi configuration. Use dot notation, for example, theme.openapi.nativeScrollbars',
```

File: packages/cli/src/index.ts (hunk 6 above)
Comment (lornajane) [reply packages/cli/src/index.ts:460]: I think we should avoid the expansion here, the content will be displayed on the terminal as help output, so shorter is better, or it causes wrapping output.

**Pull request merged**